### PR TITLE
Editorial changes, to fix the example and update security considerations

### PR DESCRIPTION
**Pull request by asingh-g**
…s other editorial changes

### CODE REVIEW

File: draft-ietf-masque-connect-udp-listen.md [hunk 1]
```diff
@@ -532,19 +540,22 @@ proxied over HTTP back to the client.
...
-
 /* Proxy confirms unmapped IP rejection. */
             <-------- CAPSULE
                         Type = COMPRESSION_CLOSE
                         Context ID = 2
-/* Proxy drops any packets received on the
-bound IP(s):Port */
+
+/* All traffic that does not belong to the registered */
+/* Context ID 4 = 223.0.2.31:4321 is dropped at the proxy */
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-udp-listen.md:556]: Wait context ID 4 should still be open here right?

Comment (asingh-g) [reply draft-ietf-masque-connect-udp-listen.md:556]: I updated it to make a bit more sense

File: draft-ietf-masque-connect-udp-listen.md [hunk 2]
```diff
@@ -373,10 +372,10 @@ review the security considerations in {{Section 21 of ?TURN=RFC8656}}.
 Since unextended UDP Proxying requests carry the target as part of the request,
 the proxy can protect unauthorized targets by rejecting requests before
 creating the tunnel, and communicate the rejection reason in response header
-fields. Bound UDP Proxying requests do not have this ability. Therefore,
-proxies MUST validate the target on every datagram and MUST NOT forward
-individual datagrams with unauthorized targets. Proxies can either silently
-discard such datagrams or abort the corresponding request stream.
+fields. Bound UDP Proxying requests, on the other hand, MUST reject all traffic
```
[the remark this answers is not in the document]
Comment (asingh-g) [reply draft-ietf-masque-connect-udp-listen.md:375]: maybe its the double negative, I'm trying to say that all traffic that doesn't belong to a bound context should be dropped

File: draft-ietf-masque-connect-udp-listen.md [hunk 3]
```diff
@@ -373,10 +372,10 @@ review the security considerations in {{Section 21 of ?TURN=RFC8656}}.
...
+on compressed contexts that does not correspond to the mapped target. On the
+other hand, an uncompressed context SHALL allow all traffic through to the
```
[the remark this answers is not in the document]
Comment (asingh-g) [reply draft-ietf-masque-connect-udp-listen.md:377]: I agree, this is better, I'll use it instead

File: draft-ietf-masque-connect-udp-listen.md [hunk 4]
```diff
@@ -493,37 +487,54 @@ proxied over HTTP back to the client.
                          UDP Port = 1234
                          UDP Payload = Encapsulated UDP Payload
 
-/* Wait for the STUN server to send the proxy's IP and */
-/* port to the other browser and for the other browser */
-/* to send a UDP packet to the proxy. */
+/ *Client responds on the same uncompressed context */
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-udp-listen.md:490]:
```suggestion
/ * Client responds on the same uncompressed context */
```

File: draft-ietf-masque-connect-udp-listen.md [hunk 5]
```diff
@@ -373,10 +372,13 @@ review the security considerations in {{Section 21 of ?TURN=RFC8656}}.
...
-proxies MUST validate the target on every datagram and MUST NOT forward
-individual datagrams with unauthorized targets. Proxies can either silently
-discard such datagrams or abort the corresponding request stream.
+fields. The uncompressed context allows transporting datagrams to and from any
+target. Clients that keep the uncompressed context open need to be able to
+receive from all targets. If the UDP proxy would reject unextended UDP proxying
+requests to some targets (as recommended in {{Section 7 of CONNECT-UDP}}), then
+for bound UDP proxying requests where the uncompressed context is open, the
+UDP proxy needs to perform checks on the target of each uncompressed context
+datagram.
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-udp-listen.md:381]:
```suggestion
datagram it receives.
```

File: draft-ietf-masque-connect-udp-listen.md [hunk 6]
```diff
@@ -493,37 +487,54 @@ proxied over HTTP back to the client.
...
                          Quarter Stream ID = 11
                          Context ID = 2
                          IP Version = 4
-                         IP Address = 203.0.113.33
+                         IP Address = 203.0.113.11
                          UDP Port = 4321
                          UDP Payload = Encapsulated UDP Payload
 
-/* Register 203.0.113.33:1234 to compress it in the future*/
+/ *Client responds on the same uncompressed context */
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-udp-listen.md:509]:
```suggestion
/ * Client responds on the same uncompressed context */
```

File: draft-ietf-masque-connect-udp-listen.md [hunk 7]
```diff
@@ -532,19 +543,22 @@ proxied over HTTP back to the client.
...
-
 /* Proxy confirms unmapped IP rejection. */
             <-------- CAPSULE
                         Type = COMPRESSION_CLOSE
                         Context ID = 2
-/* Proxy drops any packets received on the
-bound IP(s):Port */
+
+/* Context ID 4 = 203.0.113.11:4321 traffic is accepted, */
+/* And the rest is dropped at the proxy*/
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-udp-listen.md:559]:
```suggestion
/* And the rest is dropped at the proxy */
```

**Review by DavidSchinazi — APPROVED** [review]
LGTM!

**Pull request merged**